src/inmanta/data/__init__.py

```diff
@@ -2427,6 +2427,17 @@ async def get_next_version(self) -> int:
         self.last_version = version
         return version
 
+    async def add_new_setting(self, setting: Setting) -> None:
+        """
+        Adds a new setting in this environment.
+
+        :param setting: the setting that should be added to the existing settings
+        """
+        assert isinstance(setting, Setting)
```

Comment:
I don't know if those checks are really needed or if something else should be checker

Reply:
Indeed. This is not really needed. If we violate this constraint, the type checker should warn us.